### PR TITLE
Create course run image from course api

### PR DESCRIPTION
**Pull request by mjfrey**
Create a course_run image from the course api url if no marketing site url exists for the given partner.

### CODE REVIEW

**Comment by mjfrey** [comment]
@mattdrayer @bderusha Can you guys take a look.
This is needed to load image URLs for WL course data since we do not have drupal.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 1]
```diff
@@ -125,6 +126,16 @@ def update_course_run(self, course, body):
 
         CourseRun.objects.update_or_create(key=body['id'], defaults=defaults)
 
+    def get_courserun_image(self, body):
+        image = None
```
Comment (mattdrayer) [inline course_discovery/apps/course_metadata/data_loaders/api.py:133]: docstring?

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 2]
```diff
@@ -148,11 +148,16 @@ def assert_course_run_loaded(self, body):
         self.assertEqual(course_run.enrollment_end, AbstractDataLoader.parse_date(body['enrollment_end']))
         self.assertEqual(course_run.pacing_type, self.loader.get_pacing_type(body))
         self.assertEqual(course_run.video, self.loader.get_courserun_video(body))
+        self.assertEqual(course_run.image, self.loader.get_courserun_image(body))
 
     @responses.activate
-    def test_ingest(self):
+    @ddt.data(True, False)
+    def test_ingest(self, marketing_url):
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: Provide a name that better explains what this does.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 3]
```diff
@@ -125,6 +126,16 @@ def update_course_run(self, course, body):
...
+        if not self.partner.marketing_site_url_root:
+            image_url = body['media']['image']['raw']
```
Comment (mattdrayer) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: Let's access this attribute the same way as get_courserun_video

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py (hunk 2 above)
Comment (mattdrayer) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: Nit: 'marketing_url_enabled' ? (or available)

File: course_discovery/apps/course_metadata/tests/mock_data.py [hunk 4]
```diff
@@ -53,7 +53,10 @@
             },
             'course_video': {
                 'uri': None
-            }
+            },
+            'image': {
+                'raw': None,
```
Comment (mattdrayer) [inline course_discovery/apps/course_metadata/tests/mock_data.py]: Let's have this completely missing instead of a None value

**Comment by clintonb** [comment]
This poses issues for edx.org. Please update the loader to only set the image if the partner has no marketing site defined. We don't want to pull that image for edx.org.

**Comment by mjfrey** [comment]
@clintonb Look at get_courserun_image I already check if the marketing site is available and only load the image if its empty.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 5]
```diff
@@ -125,6 +126,16 @@ def update_course_run(self, course, body):
 
         CourseRun.objects.update_or_create(key=body['id'], defaults=defaults)
 
+    def get_courserun_image(self, body):
+        image = None
+        if not self.partner.marketing_site_url_root:
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: This will result in all images being removed for edx.org.

**Comment by mjfrey** [comment]
@clintonb never mind I see

**Comment by mattdrayer** [comment]
LGTM -- 👍

**Comment by mjfrey** [comment]
@clintonb another look.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 6]
```diff
@@ -122,9 +122,22 @@ def update_course_run(self, course, body):
             'video': self.get_courserun_video(body),
             'pacing_type': self.get_pacing_type(body),
         }
+        if not self.partner.marketing_site_url_root:
+            image = self.get_courserun_image(body)
+            if image:
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: Remove this. If the image is, for some reason, removed, we should remove it from our data.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 7]
```diff
@@ -122,9 +122,22 @@ def update_course_run(self, course, body):
             'video': self.get_courserun_video(body),
             'pacing_type': self.get_pacing_type(body),
         }
+        if not self.partner.marketing_site_url_root:
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/api.py:127]: Please add a comment explaining why this is necessary.

**Comment by clintonb** [comment]
👍 (after comments addressed)

**Pull request merged**